### PR TITLE
Hide push notifications from merged branches

### LINKED ISSUE #25778: Non-sense "You pushed on branch ...." notification
### Description

Reproduce step:

1. Create a branch
2. Create a PR
3. Merge the PR
4. Go to the repo home

Now, these 2 notifications are non-sense.

* [x] For "main" branch: yes, it is changed.
* [ ] For "root-patch-1": it has been merged.
 
image

### Gitea Version

1.21

### PR DESCRIPTION
**Pull request by dsseng**
Fixes #25778

### CODE REVIEW

**Comment by yp05327** [comment]
How about:
``` golang
Where(builder.And(
    builder.Eq{"pull_request.head_repo_id": repoID,},
    builder.Or(
        builder.Eq{"issue.is_closed": false,},
        builder.Eq{"pull_request.has_merged": true,},
    ),
))
```

**Comment by sh7dm** [comment]
@yp05327 looks great!

**Comment by yp05327** [comment]
@lunny 
I have a question which is not related to this PR, but maybe it can be done in this PR:
If user created a PR, and closed it manually, this `create new pr` notification will show again which seems strange.

Think about this:
I have created a branch to do something, e.g. a bug fix, and created a PR.
But I found that this bug already fixed by another PR, so I closed this PR.
Then in the home page of this repo, you will see this notification again until you delete the branch or wait for 6h.
And GitHub will not display this notification again if user closed PR manually.

Maybe we can remove this: `builder.Eq{"issue.is_closed": false,},`

**Comment by lunny** [comment]
> @lunny I have a question which is not related to this PR, but maybe it can be done in this PR: If user created a PR, and closed it manually, this `create new pr` notification will show again which seems strange.
> 
> Think about this: I have created a branch to do something, e.g. a bug fix, and created a PR. But I found that this bug already fixed by another PR, so I closed this PR. Then in the home page of this repo, you will see this notification again until you delete the branch or wait for 6h. And GitHub will not display this notification again if user closed PR manually.
> 
> Maybe we can remove this: `builder.Eq{"issue.is_closed": false,},`

In that case, I think you should delete the branch then it will not be pop up.

File: models/git/branch.go [hunk 1]
```diff
@@ -387,10 +387,13 @@ func FindRecentlyPushedNewBranches(ctx context.Context, repoID, userID int64, ex
...
 		InnerJoin("issue", "issue.id = pull_request.issue_id").
-		Where(builder.Eq{
-			"pull_request.head_repo_id": repoID,
-			"issue.is_closed":           false,
-		})
+		Where(builder.And(
+			builder.Eq{"pull_request.head_repo_id": repoID},
+			builder.Or(
+				builder.Eq{"issue.is_closed": false},
+				builder.Eq{"pull_request.has_merged": true},
```
Comment (delvh) [inline models/git/branch.go:394]: Wait, that should be `false`, right?

**Comment by lunny** [comment]
replaced by #25812

**Pull request closed without merging**